spec/rspec/mocks/verifying_doubles/construction_spec.rb

```diff
@@ -60,6 +62,43 @@ module CustomModule
 
           instance_double(test_class, :some_method_defined_dynamically => true)
         end
+
+        describe 'any_instance' do
+          let(:test_class) { Class.new(ClassThatDynamicallyDefinesMethods) }
+          before(:each) do
+            RSpec.configuration.mock_with(:rspec) do |config|
+              config.verify_partial_doubles = true
+              config.when_declaring_verifying_double do |reference|
+                reference.target.define_attribute_methods! if reference.target.is_a?(Class)
+              end
+            end
+          end
+
+          it 'calls the callback for expect_any_instance_of' do
+            expect_any_instance_of(test_class).to receive(:some_method_defined_dynamically)
+            expect { expect_any_instance_of(test_class).to receive(:some_invalid_method) }.
```

Comment:
Typically we format such invocations like this:
```
expect {
  expect_any_instance_of(test_class).to receive(:some_invalid_method)
}.to raise_error(RSpec::Mocks::MockExpectationError, ...)
```

I think I'd prefer that, and assign the message to a let given its the same...